pkg/disk/partition_table.go

```diff
@@ -373,6 +373,7 @@ func (pt *PartitionTable) relayout(size uint64) uint64 {
 			rootIdx = idx
 			continue
 		}
+		start += partition.Start
```

Comment:
Let me comment on this that this change was suggested by me. Normally `PartitionTable` will find the start offset by itself based on the header and partition sizes. This adds the `Start` attribute if it already had a value (otherwise it will be `0`) to correctly offset the partitions (and keeps track of it since it happens in the loop).

I've built a manifest and image with these changes and the result is:
```
user@muja image € fdisk -l disk.img
Disk disk.img: 3.69 GiB, 3965714432 bytes, 7745536 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: D209C89E-EA5E-4FBD-B161-B461CCE297E0

Device       Start     End Sectors  Size Type
disk.img1    18432  428031  409600  200M EFI System
disk.img2   428032 1452031 1024000  500M Linux extended boot
disk.img3  1452032 7745502 6293471    3G Linux filesystem
```

However, this change *does* hit all partition table generation and as such please do take a good look at this change too :)

Reply:
we may still have a bug here, I will verify Monday it's not my testing process.

Reply:
It was my testing process, it works for our offset needs. That concern is resolved for me, but I won't resolve the conversation due to the need of wider review.

Reply:
I think this change makes things hard to follow and while it does have the intended effect when combined with the next commit, it's doing it in a way that could lead to bugs or confusing behaviour in the future.

The `partition.Start` is meant to be calculated from the partition table generator based on the order of partitions and their size.  With this change, any `Start` defined in the static partition table becomes essentially an offset or gap ahead of the partition.  Also, it's ignored for `/`.

The next commit in the PR adds a `Start` value to the first partition in the minimal raw partition tables which has the intended effect but consider that:
1. If, for whatever reason, the first partition in a table is `/`, any start/offset, wont be applied.
2. If, for whatever reason, we ever need to add gaps between partitions, it wont be added for `/`.  More importantly, the desire for this is questionable but it is something we're enabling with this change.
3. Even if we do ever want to add gaps between partitions, I'd rather add a new field rather than give the `Start` field the double meaning of "offset/gap" (when defined on the static partition table) and "absolute start position" (after processing).

Since the purpose of this change is to add a start offset for the entire PT, I propose we add a new field to the `PartitionTable` struct called `Offset` (or `StartOffset`), like we do for `ExtraPadding`: https://github.com/osbuild/images/blob/4f0f01389dec364f151b4b96f840dd2aa9dedeba/pkg/disk/partition_table.go#L19
and simply add that to the `header` after it's calculated at the top of the `relayout()` function.

Reply:
This was based on what you had done in commit 48466ae44. From my experience at least there has been numerous occasions to define an offset for the start of all partitions. EG all vmware virt images with certain releases of VMWare used to need to start at a particular offset due to enterprise arrays and how they used to deal with blocks. I've never see a use/need of similarly specific spaces between concurrent partitions, only ever at the very beginning.

Reply:
I've replaced this with a patch similar to the extrapadding.